Formula/trojan-go.rb

```diff
@@ -0,0 +1,114 @@
+class TrojanGo < Formula
+  desc "Trojan proxy in Go"
+  homepage "https://p4gefau1t.github.io/trojan-go/"
+  url "https://github.com/p4gefau1t/trojan-go.git",
+      tag:      "v0.10.0",
+      revision: "a9ce6d4ada9143d58e8d1373d23f9ebfe871119e"
+  license "GPL-3.0-only"
+  head "https://github.com/p4gefau1t/trojan-go.git"
+
+  depends_on "go" => :build
+
+  resource "geoip" do
+    geoip_hash_url = "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat.sha256sum"
+    url "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat"
+    sha256 Utils.safe_popen_read("curl", "-L", geoip_hash_url)[0, 64]
+  end
+
+  resource "geosite" do
+    geosite_hash_url = "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat.sha256sum"
+    url "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat"
+    sha256 Utils.safe_popen_read("curl", "-L", geosite_hash_url)[0, 64]
+  end
+
+  def install
+    commit = Utils.git_head
+    tag = Utils.safe_popen_read("git", "tag", "--contains", "HEAD").strip
+    execpath = libexec / name
+    ldflags = %W[
+      -X github.com/p4gefau1t/trojan-go/constant.Version=#{tag}
+      -X github.com/p4gefau1t/trojan-go/constant.Commit=#{commit}
+    ]
+
+    system "go", "build", *std_go_args, "-o", execpath, "-tags=full", "-ldflags", ldflags.join(" ")
+    (bin / "trojan-go").write_env_script execpath,
+      TROJAN_GO_LOCATION_ASSET: "${TROJAN_GO_LOCATION_ASSET:-#{pkgshare}}"
+
+    pkgetc.install "example/client.json" => "config.json"
+
+    resource("geoip").stage do
+      pkgshare.install "geoip.dat"
+    end
+
+    resource("geosite").stage do
+      pkgshare.install "dlc.dat" => "geosite.dat"
+    end
+  end
+
+  plist_options manual: "trojan-go -config #{HOMEBREW_PREFIX}/etc/trojan-go/config.json"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+        <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{bin}/trojan-go</string>
+            <string>-config</string>
+            <string>#{etc}/trojan-go/config.json</string>
+          </array>
+          <key>KeepAlive</key>
+          <true/>
+          <key>RunAtLoad</key>
+          <true/>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    output = shell_output "#{bin}/trojan-go -version"
+    assert_match "Trojan-Go", output
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
I am afraid I do not know how to write it.
This can hardly be done without requiring upstream code modification.

Reply:
It fails because there is no http server and the cert is invalid.

Reply:
Ok, let's try running `trojan-go` without the `fork` + `exec` and check that it returns the expected non-zero exit code and expected error message.